homework-submission/homework-week1/homework-week2/App.js

```diff
@@ -0,0 +1,143 @@
+const ListItem = ({ id, description, deadline, clickHandler, className }) => {
+  return (
+    <li key={id} id={id} onClick={clickHandler} className={className}>
+      {description} , {deadline}
+    </li>
+  );
+};
+
+const Button = ({ type, text, className }) => {
+  return (
+    <button type={type} className={className}>
+      {text}
+    </button>
+  );
+};
+
+const DynamicList = ({ List, clickHandler, className }) => {
+  return (
+    <div className="list">
+      {List.map(item => {
+        return (
+          <ListItem
+            clickHandler={clickHandler}
+            key={item.id}
+            id={item.id}
+            description={item.description}
+            deadline={item.deadline}
+            className={className}
+          />
+        );
+      })}
+    </div>
+  );
+};
+
+const AddingListItem = ({ submitHandler }) => {
+  let today = new Date();
+  const dd = String(today.getDate()).padStart(2, '0');
+  const mm = String(today.getMonth() + 1).padStart(2, '0');
+  const yyyy = today.getFullYear();
+  today = yyyy + '-' + mm + '-' + dd;
+  return (
+    <form onSubmit={submitHandler}>
+      <input type="text" name="listItem" defaultValue={'Do something , ' + today} />
+      <Button type="submit" text="Add Item" className="addItem" />
+    </form>
+  );
+};
+
+class App extends React.Component {
+  state = {
+    todoList: [
+      {
+        id: 0,
+        description: 'Get out of bed',
+        deadline: '2017-09-11',
+      },
+      {
+        id: 1,
+        description: 'Brush teeth',
+        deadline: '2017-09-10',
+      },
+      {
+        id: 2,
+        description: 'Eat breakfast',
+        deadline: '2017-09-09',
+      },
+    ],
+  };
+
+  addItem = event => {
```

Comment:
There are a lot of things happening in the function. Ideally a function should do one thing. So could be an idea to split it up in smaller functions which you call in this one.